modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/DataStreamIT.java

```diff
@@ -1779,7 +1779,14 @@ public void testRemoveGhostReference() throws Exception {
                 public ClusterState execute(ClusterState currentState) throws Exception {
                     DataStream original = currentState.getMetadata().dataStreams().get(dataStreamName);
                     DataStream broken = original.copy()
-                        .setIndices(List.of(new Index(original.getIndices().get(0).getName(), "broken"), original.getIndices().get(1)))
+                        .setBackingIndices(
+                            original.getBackingIndices()
+                                .copy()
+                                .setIndices(
+                                    List.of(new Index(original.getIndices().get(0).getName(), "broken"), original.getIndices().get(1))
+                                )
+                                .build()
+                        )
```

Comment:
I think providing this helper method makes sense - though since it's internal to the DataStreamIndices record and it's creating a builder just to immediately build the result, should we just skip the builder in the method and make the new record manually?

Reply:
I used the `copy()` method in my suggestion snippet to avoid having to copy all the fields in the record construction. After all, that's the value of the `copy()` method, right?

Reply:
Hmm if I create a helper method like in my snippet, I only save two lines (`.copy()` & `.build()`) in the lines above. That's very minimal, so not sure whether it's worth the extra helper method. What do you think?

Reply:
I was originally thinking it would save us from making and immediately throwing away a builder instance but I don't think it's worth worrying about